tests/test_cli_client.cpp

```diff
@@ -1390,6 +1401,35 @@ TEST_F(Client, set_handles_persistent_settings_errors)
     EXPECT_THAT(send_command({"set", keyval_arg(key, val)}), Eq(mp::ReturnCode::CommandFail));
 }
 
+TEST_F(Client, get_returns_acceptable_autostart_value_by_default)
+{
+    EXPECT_THAT(get_setting(mp::autostart_key), AnyOf("true", "false"));
+}
+
+TEST_F(Client, set_cmd_rejects_bad_autostart_value)
+{
+    aux_set_cmd_rejects_bad_val(mp::autostart_key, "asdf");
+    aux_set_cmd_rejects_bad_val(mp::autostart_key, "True");
+    aux_set_cmd_rejects_bad_val(mp::autostart_key, "trueasdf");
+    aux_set_cmd_rejects_bad_val(mp::autostart_key, "False");
+    aux_set_cmd_rejects_bad_val(mp::autostart_key, "0");
+    aux_set_cmd_rejects_bad_val(mp::autostart_key, "1");
+    aux_set_cmd_rejects_bad_val(mp::autostart_key, "123");
+    aux_set_cmd_rejects_bad_val(mp::autostart_key, "");
```

Comment:
OK, I can do that.

Reply:
I made it support "true", "on", "yes", "1", "false", "off", "no", "0", and all upper case variations thereof. We can only cover so much natural language, so I use an explicit conversion and  still reject everything else and, rather than giving the input directly to Qt and have it interpreting non-empty strings (e.g. "-", "nope", "n", "000", "nein") as true.

Reply:
Yeah totally. Could you make it case insensitive?

Reply:
Sure, I did already.

Reply:
Oh, I misunderstood "upper case variations" :)